qutip/tests/test_random.py

```diff
@@ -71,13 +72,34 @@ def check_func_dims(func, args, kwargs, dims):
     # TODO: promote this out of test_random, as it's generically useful
     #       in writing tests.
     resdims = func(*args, **kwargs).dims
-    assert_(resdims == dims, "Checking {}; epected dimensions of {}, got {}.".format(func.__name__, dims, resdims))
+    assert_(resdims == dims, "Checking {}; expected dimensions of {}, got {}.".format(func.__name__, dims, resdims))
+
+def check_func_shape(func, args, kwargs, shape):
+    resdims = func(*args, **kwargs).dims
+    assert_(dims_to_tensor_shape(resdims)==shape,"Checking {}; expected shape of {}, got {}.".format(func.__name__, shape, dims_to_tensor_shape(resdims)))
```

Comment:
I don't understand the purpose of this test?  `dims_to_tensor_shape` isn't related to what you're trying to do here; it's for resolving some reorderings that superoperators need.  You're already testing the dimensions in `check_func_dims`, and `dims_to_tensor_shape` in this case will always just return the dimensions as a "flattened" tuple.

Was there another reason for including it?

Reply:
For some reason, I remember thinking that I should test the tuple as well. Unfortunately, I forgot that exact reason ! Like you said, the tensor structure is actually tested in `check_func_dims`. I think that's why I was confused about ways to test tensor structure. I will make sure to remove this check.